### ISSUE TITLE
Debugger: Collapse multiline values in variable pane preview

### ISSUE THREAD

**Issue by rehmsen** [microsoft/vscode#212167]
In the debugger variable pane, VSCode shows a truncated, single line preview of the value returned in the DAP Variable type. It would often be more useful to show more than just the first line, and it is further confusing that when the first line is shorter than the width provided, no ellipsis is shown, so that it is unclear that there is actually a second line.

For example, if the DAP value is something like:
```
{
  "name":"city",
  "value":"criterion_id: 1\ncountry_code: \"DE\"\ncity_name: \"chicago\"\n"
  ...
}
```

Currently we show this: 
![just "criterion: 1"](https://github.com/microsoft/vscode/assets/29040/40c562f1-0ab6-4ce5-a45a-50c90fe95f19)
 
I think it would be more useful to not use `white-space: pre`, so that instead we show:
![image](https://github.com/microsoft/vscode/assets/29040/ad2d3dba-4234-4f1c-ba1a-ecc397f0de97)

First, if there is more space, it is used. Second, there is always an ellipsis symbol when something was cut off, not just if there is something cut off in the first line.

Happy to send a PR deleting the `white-space: ellipsis` if you agree this is an improvement.

**Comment by roblourens** [comment]
I'd like this, but we should show the newlines in the collapsed value

**Comment by rehmsen** [comment]
Good idea, we could use the ⏎ for that. But I don't think that can be done in CSS only. I won't get to it this week, but could send a PR next week.

**Comment by rehmsen** [comment]
Sorry for not getting back here. I was not able to prioritize this in the end. Our immediate pain point is addressed with a CSS only patch on our fork. I agree it would be nicer to have the newline signs but I cannot make time for it at the moment to build and upstream that.
